packages/protocol/contracts/L1/verifiers/PseZkVerifier.sol

```diff
@@ -42,12 +53,29 @@ contract PseZkVerifier is EssentialContract, IVerifier {
         // Do not run proof verification to contest an existing proof
         if (isContesting) return;
 
-        bytes32 instance = getInstance(prover, evidence);
+        ProofData memory data = abi.decode(evidence.proof, (ProofData));
+        bytes32 instance = getInstance(prover, data.txListHash, evidence);
+
+        // Verify blob
+
+        // Question: What if x is larger than the blob data size
+
+        // Question: how to calculate pointCommitment and pointProof
+        // offchain?
```

Comment:
It's a standard KZG commitment I think so should be plenty of libraries available to do this from the data. The proof does also depend on the randomness, so having it calculated directly by the prover next to the actual proof seems to make sense to me.